### PR TITLE
[release-v1.9] Add github action to validate

### PR DESCRIPTION
**Pull request by nak3**
This adds github action for validation.

This also changes the version label. The `1.9.0` is not accurate actually so I think it is alright but please any suggestions are welcome.

### CODE REVIEW

**Review by ReToCode — APPROVED** [review]
Nice!

/lgtm
/approve

/hold for other reviewers to chime in.

File: openshift/release/artifacts/serving-post-install-jobs.yaml [hunk 1]
```diff
@@ -34,7 +34,7 @@ spec:
         app: storage-version-migration-serving
         app.kubernetes.io/name: knative-serving
         app.kubernetes.io/component: storage-version-migration-job
-        app.kubernetes.io/version: "v1.9.0"
+        app.kubernetes.io/version: "release-v1.9"
```
Comment (skonto) [inline openshift/release/artifacts/serving-post-install-jobs.yaml:37]: Was this fixed manually?

File: openshift/release/generate-release.sh [hunk 2]
```diff
@@ -19,10 +19,6 @@ readonly SERVING_POST_INSTALL_JOBS_YAML=${YAML_OUTPUT_DIR}/serving-post-install-
 if [[ "$VERSION" == "ci" ]]; then
   # Do not use devel as operator checks the version.
   VERSION="v1.2.0"
-else
```
Comment (skonto) [inline openshift/release/generate-release.sh:22]: Was this never used?

**Comment by skonto** [comment]
Nice, lgtm, added a few questions to clarify the changes beyond the github action (which btw runs ok).

File: openshift/release/artifacts/serving-post-install-jobs.yaml (hunk 1 above)
Comment (nak3) [reply openshift/release/artifacts/serving-post-install-jobs.yaml:37]: It was generated by `openshift/release/generate-release.sh "BRANCH_NAME"` so `release-v1.9` is the branch name.

File: openshift/release/generate-release.sh (hunk 2 above)
Comment (nak3) [reply openshift/release/generate-release.sh:22]: `create-release-branch.sh` has an issue and so fixed it and added this condition again.

**Comment by nak3** [comment]
I guess https://github.com/openshift-knative/serving/pull/209 needs to be cherry-picked for 1.9 branch.

**Comment by ReToCode** [comment]
> I guess https://github.com/openshift-knative/serving/pull/209 needs to be cherry-picked for 1.9 branch.

I think also to 1.8, right? @skonto

**Comment by skonto** [comment]
> I guess https://github.com/openshift-knative/serving/pull/209 needs to be cherry-picked for 1.9 branch.

As discussed earlier will do the patches for 1.9 ;)

**Comment by skonto** [comment]
@nak3 could you rebase?

**Comment by nak3** [comment]
prow job runs job after applying the patch into the latest commits so it should be alright without rebase as long as no confliction.

[log](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift-knative_serving/201/pull-ci-openshift-knative-serving-release-v1.9-412-e2e-aws-ocp-412/1638709839418888192/build-log.txt)
```
[36mINFO[0m[2023-03-23T01:10:55Z] Resolved source https://github.com/openshift-knative/serving to release-v1.9@15a82571, merging: #201 dabe9817 @nak3 
```

**Pull request merged**